services/core-web/src/components/modalContent/ViewNoticeOfDepartureModal.js

```diff
@@ -44,25 +54,30 @@ export const ViewNoticeOfDepartureModal = (props) => {
     (doc) => doc.document_type !== NOTICE_OF_DEPARTURE_DOCUMENT_TYPE.CHECKLIST
   );
 
+  useEffect(() => {
+    const statuses = (() => {
+      const { self_authorized, ...coreStatuses } = NOTICE_OF_DEPARTURE_STATUS_VALUES;
+      return Object.values(coreStatuses);
+    })();
+
+    setStatusOptions(
+      statuses.map((status) => {
+        return {
+          value: status,
+          label: NOTICE_OF_DEPARTURE_STATUS[status],
+        };
+      })
+    );
+  }, []);
+
   const handleDeleteANoticeOfDepartureDocument = async (document) => {
     await removeFileFromDocumentManager(document);
 
     await props.fetchDetailedNoticeOfDeparture(mine.mine_guid, nod_guid);
   };
 
-  const statuses = (() => {
-    const { self_authorized, ...coreStatuses } = NOTICE_OF_DEPARTURE_STATUS_VALUES;
-    return Object.values(coreStatuses);
-  })();
-
-  // eslint-disable-next-line no-return-assign
-  const handleChangeNodStatus = ({ value }) => (props.noticeOfDeparture.nod_status = value);
-
-  const updateNoticeOfDepartureStatus = async () => {
-    await props.updateNoticeOfDeparture(
-      { mineGuid: mine.mine_guid, nodGuid: nod_guid },
-      noticeOfDeparture
-    );
+  const updateNoticeOfDepartureSubmit = async (values) => {
+    await props.updateNoticeOfDeparture({ mineGuid: mine.mine_guid, nodGuid: nod_guid }, values);
```

Comment:
nitpick. probably a good idea to fully destructure props for the sake of consistency.

Reply:
That's actually a quirk of redux.  It raises an error if you try to destructure the methods passed in through redux due to it seeing them as shadowed properties.  When I was first working on the forms I played with a few different options, and this felt like the least ugly of them. 🤷

Reply:
I think that's there to prevent you from trying to call the redux methods directly from an import instead of wiring them up through the `connect` function in the export.

Reply:
Okay yes. make sense